### PR TITLE
Fix graph modes

### PR DESCRIPTION
**Pull request by ret2libc**

### CODE REVIEW

File: libr/core/graph.c [hunk 1]
```diff
@@ -3165,23 +3165,12 @@ static int agraph_refresh(struct agraph_refresh_data *grd) {
...
 		char *title = get_title (addr);
 
-		if (!acur || strcmp (acur->title, title)) {
+		if (!acur || strcmp (acur->title, title) != 0) {
 			r_core_cmd0 (core, "sr PC");
 		}
 		free (title);
 		g->is_instep = false;
 	}
-	if (r_config_get_i (core->config, "graph.offset")) {
```
Comment (radare) [inline libr/core/graph.c]: this breaks graph.offset

File: libr/core/graph.c [hunk 2]
```diff
@@ -3663,9 +3652,6 @@ R_API int r_core_visual_graph(RCore *core, RAGraph *g, RAnalFunction *_fcn, int
 			r_config_hold_free (hc);
 			return false;
 		}
-		if (r_config_get_i (core->config, "graph.offset")) {
```
Comment (radare) [inline libr/core/graph.c]: this breaks graph.offset

**Comment by radare** [comment]
nope. this pr makes graph.offset useless again

File: libr/core/graph.c [hunk 3]
```diff
@@ -3165,23 +3165,12 @@ static int agraph_refresh(struct agraph_refresh_data *grd) {
 		addr = r_core_anal_get_bbaddr (core, addr);
 		char *title = get_title (addr);
 
-		if (!acur || strcmp (acur->title, title)) {
+		if (!acur || strcmp (acur->title, title) != 0) {
```
[the remark this answers is not in the document]
Comment (ret2libc) [reply libr/core/graph.c]: yes it was just part of the revert commit, i'm going to fix it

**Comment by ret2libc** [comment]
It doesn't break graph.offset, it fixes it and it fixes graph mode. Currently you cannot switch between different graph modes anymore because of recent commits. This PR fixes graph.offset and graph modes.

To make graph.offset useful you don't need to switch graph mode, you just need to enable asm.offset IMO, as done at https://github.com/radare/radare2/pull/10000/files#diff-eec787b9f6e2907ed4274fe29b5a02f6R1959

**Comment by ITAYC0HEN** [comment]
Mazel Tov for being issue #10000

**Comment by ret2libc** [comment]
@ITAYC0HEN Thank you :-)

**Pull request merged**